fairlearn/metrics/_annotated_metric_function.py

```diff
@@ -48,8 +48,8 @@ def __init__(
         *,
         func: Callable,
         name: Optional[str],
-        positional_argument_names: List[str] = None,
-        kw_argument_mapping: Dict[str, str] = None,
+        positional_argument_names: Optional[List[str]] = None,
+        kw_argument_mapping: Optional[Dict[str, str]] = None,
```

Comment:
I prefer this:

```suggestion
        positional_argument_names: List[str] | None = None,
        kw_argument_mapping: Dict[str, str] | None = None,
```

you might need a `from __future__ import annotations` at the start of the file

Reply:
Oh that would be great actually ! :ok_hand: 

I can by the same occasion get rid of the `Dict`, `List` type hints and use plain standard classes `dict`, `list` etc .. (only for the metrics module for now to limit the scope of this PR :sweat_smile: )

Reply:
@adrinjalali Done :exploding_head: